src/helpers/reducers/filter-reducer.js

```diff
@@ -0,0 +1,62 @@
+const initial = {
+    sortByPrice: '',
+    range: '0',
+    outOfStock: false,
+    fastDelivery: false,
+    category: [],
+    ratings: 1,
+}
+
+const filtersReducer = (filters, action) => {
+    switch (action.type) {
+        case 'HIGH_TO_LOW':
+            return {
+                ...filters,
+                sortByPrice: action.payload,
+            }
+        case 'LOW_TO_HIGH':
+            return {
+                ...filters,
+                sortByPrice: action.payload,
+            }
+        case 'PRICE_RANGE':
+            return {
+                ...filters,
+                range: action.payload,
+            }
+        case 'OUT_OF_STOCK':
+            return {
+                ...filters,
+                outOfStock: action.payload,
+            }
+        case 'FAST_DELIVERY':
+            return {
+                ...filters,
+                fastDelivery: action.payload,
+            }
+        case 'INCLUDE_CATEGORY':
+            return action.from === 'home'
+                ? { ...filters, category: [action.payload] }
```

Comment:
In my opinion, it would be more clearer to use simple if else statements when there's such nesting of ternary operator.

Reply:
if else makes the code fat. 
Switch statements trim the fat plus are easier to analyze.